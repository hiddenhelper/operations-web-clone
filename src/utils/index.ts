export * from './generalUtils';
export * from './useValidator';
