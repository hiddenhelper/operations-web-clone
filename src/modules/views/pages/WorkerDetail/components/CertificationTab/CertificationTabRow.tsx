import React, { memo, useCallback } from 'react';
import { TableCell, TableRow, withStyles } from '@material-ui/core';

import TableCellLink from 'modules/views/shared/TableCellLink';

import { CertificationModel, GeneralModel, UserModel } from 'modules/models';
import { tableGlobalStyles, tableRowStyles } from 'assets/styles/Tables/styles';
import { getDefaultValue, getFormattedDate } from 'utils';
import ThreeDotsPopover from '../Shared/ThreeDotsPopover';
import PermissionGuard from 'modules/views/shared/PermissionGuard';

const StyledTableRow = withStyles(tableRowStyles)(TableRow);

export interface ICertificationRowProps {
  certification: CertificationModel.IWorkerCertification;
  openDetail: (id: string) => void;
  onEdit: (id: string) => void;
  onDelete: (id: string) => void;
}

const CertificationTabRow = ({ certification, openDetail, onEdit, onDelete }: ICertificationRowProps) => {
  const tableGlobalClasses = tableGlobalStyles();
  const openDetailHandler = useCallback(() => openDetail(certification.id), [openDetail, certification.id]);

  return (
    <StyledTableRow style={{ cursor: 'pointer' }} onClick={openDetailHandler} data-testid="certification-list-row">
      <TableCell>{certification.certification.name}</TableCell>
      <TableCell>{certification.idNumber ? certification.idNumber : '-'}</TableCell>
      <TableCell data-testid="row-completion">{getDefaultValue(getFormattedDate(certification.completionDate, GeneralModel.DateFormat.DATE_TEXT))}</TableCell>
      <TableCell data-testid="row-expiration">{getDefaultValue(getFormattedDate(certification.expirationDate, GeneralModel.DateFormat.DATE_TEXT))} </TableCell>
      <TableCell className={`${tableGlobalClasses.noTopCell} ${tableGlobalClasses.threeDotsWrapper}`} data-testid="row-project">
        {certification.project?.id ? (
          <TableCellLink href={`/projects/detail/${certification.project.id}`} text={certification.project.name} title="View Project detail" />
        ) : (
          'No Related Project'
        )}
      </TableCell>
      <TableCell data-testid="update-delete-cell">
        <PermissionGuard permissionsExpression={UserModel.WorkerCertificationsPermission.MANAGE}>
          <ThreeDotsPopover entity={certification} onEdit={onEdit} onDelete={onDelete} />
        </PermissionGuard>
      </TableCell>
    </StyledTableRow>
  );
};

export default memo(CertificationTabRow);
