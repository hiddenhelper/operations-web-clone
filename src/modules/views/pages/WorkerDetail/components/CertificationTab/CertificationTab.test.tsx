import React from 'react';
import { render, fireEvent, act } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import { createMockStore } from 'redux-test-utils';

import {
  getWorkerCertification_1,
  getWorkerCertification_2,
  getWorkerCertification_3,
  getWorker_1,
  getCertification_1,
  getCertification_2,
  getProject_1,
  getProject_2,
  getUploadFile_3,
  getWorkerProject_1,
} from '../../../../../../test/entities';
import { getInitialState } from '../../../../../../test/rootState';
import CertificationTab, { ICertificationTabProps } from './CertificationTab';
import { noop } from '../../../../../../utils/generalUtils';

describe('CertificationTab', () => {
  let props: ICertificationTabProps;

  beforeEach(() => {
    props = {
      worker: getWorker_1(),
      certificationMap: {
        [getWorkerCertification_1().id]: getWorkerCertification_1(),
      },
      fileMap: {},
      projectWorkerMap: { [getWorker_1().id]: { [getWorkerProject_1().id]: getWorkerProject_1() } },
      certificationList: [getCertification_1(), getCertification_2()],
      projectList: [getProject_1(), getProject_2()],
      queryParams: { page: 1, limit: 30 },
      count: 1,
      listLoading: {
        isLoading: false,
        hasError: false,
        error: undefined,
      },
      saveLoading: {
        isLoading: false,
        hasError: false,
        error: undefined,
      },
      detailLoading: {
        isLoading: false,
        hasError: false,
        error: undefined,
      },
      deleteLoading: {
        isLoading: false,
        hasError: false,
        error: undefined,
      },
      loadingMap: {},
      onPageChange: jest.fn(),
      addWorkerCertification: jest.fn(),
      updateWorkerCertification: jest.fn(),
      deleteWorkerCertification: jest.fn(),
      fetchCertificationList: jest.fn(),
      fetchWorkerCertificationList: jest.fn(),
      fetchWorkerCertificationDetail: jest.fn(),
      fetchProjectList: jest.fn(),
      addCertificationSuccess: jest.fn(),
      updateCertificationSuccess: jest.fn(),
      clearLoading: jest.fn(),
      clearWorkerMap: jest.fn(),
      fetchWorker: jest.fn(),
      fetchWorkerProjectList: jest.fn(),
      setQueryParams: jest.fn(),
      clearFileMap: jest.fn(),
    };
  });

  it('should fetchWorkerCertificationList', () => {
    props.certificationMap = {};
    render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <CertificationTab {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(props.fetchWorkerCertificationList).toHaveBeenCalledWith(getWorker_1().id, { limit: 30, page: 1 });
  });

  it('should fetchCertificationList', () => {
    props.certificationList = [];
    render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <CertificationTab {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(props.fetchCertificationList).toHaveBeenCalledWith();
  });

  it('should render', () => {
    const { container } = render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <CertificationTab {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with secondary values', () => {
    props.certificationMap = {
      [getWorkerCertification_1().id]: getWorkerCertification_1(),
      [getWorkerCertification_2().id]: getWorkerCertification_2(),
      [getWorkerCertification_3().id]: getWorkerCertification_3(),
    };
    const wrapper = render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <CertificationTab {...props} />
        </MemoryRouter>
      </Provider>
    );

    expect(wrapper.getAllByTestId('row-completion')[0].textContent).toEqual('January 01, 2001');
    expect(wrapper.getAllByTestId('row-completion')[1].textContent).toEqual('-');

    expect(wrapper.getAllByTestId('row-expiration')[0].textContent).toEqual('January 01, 2001 ');
    expect(wrapper.getAllByTestId('row-expiration')[1].textContent).toEqual('- ');

    expect(wrapper.container).toMatchSnapshot();
  });

  it('should render loading', () => {
    props.listLoading.isLoading = true;
    const { container } = render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <CertificationTab {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render detail loading', () => {
    props.detailLoading.isLoading = true;
    const { container, getByTestId } = render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <CertificationTab {...props} />
        </MemoryRouter>
      </Provider>
    );
    const openDetailBtn = getByTestId('certification-list-row');
    act(() => {
      fireEvent.click(openDetailBtn);
    });
    expect(container).toMatchSnapshot();
  });

  it('should render modal', () => {
    const { getByTestId } = render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <CertificationTab {...props} />
        </MemoryRouter>
      </Provider>
    );

    const openModalBtn = getByTestId('open-certification-modal-btn');
    act(() => {
      fireEvent.click(openModalBtn);
    });

    const closeModalBtn = getByTestId('assign-btn-close');

    act(() => {
      fireEvent.click(closeModalBtn);
    });
  });

  it('should add certification', () => {
    props.loadingMap = { saveWorker: { traceId: 'someOtherKey' } as any };
    props.fileMap = {
      ['workerCertification']: {
        [getUploadFile_3().id]: getUploadFile_3(),
      },
    };
    const { getByText, getByTestId, getAllByText } = render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <CertificationTab {...props} />
        </MemoryRouter>
      </Provider>
    );

    const openModalBtn = getByTestId('open-certification-modal-btn');
    act(() => {
      fireEvent.click(openModalBtn);
    });

    const addBtn = getByTestId('assign-btn-confirm');

    const certificationSelect = getAllByText('Select Option')[0];
    const projectSelect = getAllByText('Select Option')[1];

    act(() => {
      fireEvent.mouseDown(certificationSelect);
    });

    act(() => {
      fireEvent.click(getByText(getCertification_2().name));
    });

    act(() => {
      fireEvent.mouseDown(projectSelect);
    });

    const projectOptionValue = getAllByText(getProject_1().name)[1];

    act(() => {
      fireEvent.click(projectOptionValue);
    });

    act(() => {
      fireEvent.change(getByTestId('certification-idNumber'), {
        persist: noop,
        target: { value: '1238912' },
      });
    });

    act(() => {
      fireEvent.click(addBtn);
    });

    expect(props.clearLoading).toHaveBeenCalled();
    expect(props.addCertificationSuccess).toHaveBeenCalled();
    expect(props.addWorkerCertification).toHaveBeenCalledWith(
      getWorker_1().id,
      {
        certificationId: getCertification_2().id,
        completionDate: null,
        description: null,
        expirationDate: null,
        id: null,
        idNumber: '1238912',
        projectId: getProject_1().id,
      },
      'workerCertification'
    );
  });

  it('should show form validations', () => {
    const { getByText, getByTestId } = render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <CertificationTab {...props} />
        </MemoryRouter>
      </Provider>
    );

    const openModalBtn = getByTestId('open-certification-modal-btn');
    act(() => {
      fireEvent.click(openModalBtn);
    });

    const addBtn = getByTestId('assign-btn-confirm');

    act(() => {
      fireEvent.click(addBtn);
    });

    expect(getByText('Please enter Certification Name.')).toBeTruthy();
  });

  describe('detail', () => {
    it('should render values', () => {
      const { getByTestId, getAllByTestId, getByText } = render(
        <Provider store={createMockStore(getInitialState())}>
          <MemoryRouter>
            <CertificationTab {...props} />
          </MemoryRouter>
        </Provider>
      );

      const certificationList = getAllByTestId('certification-list-row');

      act(() => {
        fireEvent.click(certificationList[0]);
      });

      expect(getByText('Worker Certification')).toBeTruthy();

      const closeDetailBtn = getByTestId('detail-modal-close-btn');

      act(() => {
        fireEvent.click(closeDetailBtn);
      });
    });

    it('should render null values', () => {
      props.certificationMap = {
        [getWorkerCertification_3().id]: getWorkerCertification_3(),
      };

      const { getByTestId, getAllByTestId, getByText } = render(
        <Provider store={createMockStore(getInitialState())}>
          <MemoryRouter>
            <CertificationTab {...props} />
          </MemoryRouter>
        </Provider>
      );

      const certificationList = getAllByTestId('certification-list-row');

      act(() => {
        fireEvent.click(certificationList[0]);
      });

      expect(getByText('Worker Certification')).toBeTruthy();

      const closeDetailBtn = getByTestId('detail-modal-close-btn');

      act(() => {
        fireEvent.click(closeDetailBtn);
      });
    });
  });

  it('should filter by project', () => {
    const wrapper = render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <CertificationTab {...props} />
        </MemoryRouter>
      </Provider>
    );

    act(() => {
      fireEvent.click(wrapper.getAllByText('All Projects')[0]);
    });

    act(() => {
      fireEvent.click(wrapper.getByText('Project Awesome'));
    });

    expect(props.setQueryParams).toHaveBeenCalled();
  });

  it('should filter by no related project', () => {
    const wrapper = render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <CertificationTab {...props} />
        </MemoryRouter>
      </Provider>
    );

    act(() => {
      fireEvent.click(wrapper.getAllByText('All Projects')[0]);
    });

    act(() => {
      fireEvent.click(wrapper.getByText('No Related Project'));
    });

    expect(props.setQueryParams).toHaveBeenCalled();
  });

  it('should render edition modal', () => {
    props = {
      ...props,
      certificationMap: {
        [getWorkerCertification_1().id]: { ...getWorkerCertification_1(), projectId: null },
      },
      updateLoading: {
        isLoading: true,
        hasError: false,
        error: undefined,
      },
    };
    const { getByTestId, getByText } = render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <CertificationTab {...props} />
        </MemoryRouter>
      </Provider>
    );

    const openEditDeleteOptions = getByTestId('popover-button');

    act(() => {
      fireEvent.click(openEditDeleteOptions);
    });

    act(() => {
      fireEvent.click(getByText('Edit'));
    });

    const idNumberInput = getByTestId('certification-idNumber');

    expect(idNumberInput.value).toBe('1892132');

    act(() => {
      fireEvent.change(idNumberInput, {
        persist: noop,
        target: { name: 'idNumber', value: '19000000' },
      });
    });

    const saveModalBtn = getByTestId('assign-btn-confirm');

    act(() => {
      fireEvent.click(saveModalBtn);
    });

    expect(props.clearLoading).toHaveBeenCalled();
    expect(props.updateCertificationSuccess).toHaveBeenCalled();
    expect(props.updateWorkerCertification).toBeCalled();

    const closeModalBtn = getByTestId('assign-btn-close');

    act(() => {
      fireEvent.click(closeModalBtn);
    });
  });

  it('should render confirmation modal if delete a certification', () => {
    const { getByTestId, getByText } = render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <CertificationTab {...props} />
        </MemoryRouter>
      </Provider>
    );

    const openEditDeleteOptions = getByTestId('popover-button');

    act(() => {
      fireEvent.click(openEditDeleteOptions);
    });

    act(() => {
      fireEvent.click(getByText('Delete'));
    });

    expect(getByText('Delete Certification?'));
    expect(getByText('Yes, Delete'));

    const confirmDeleteButton = getByTestId('modal-confirm-btn');

    act(() => {
      fireEvent.click(confirmDeleteButton);
    });
  });
});
