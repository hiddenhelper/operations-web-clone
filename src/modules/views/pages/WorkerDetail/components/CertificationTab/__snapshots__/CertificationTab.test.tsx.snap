// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CertificationTab should render 1`] = `
<div>
  <div
    class="makeStyles-tableWrapper-219"
  >
    <div
      class="makeStyles-filterActionsContainer-230 makeStyles-filterActionsContainerPadding-232"
    >
      <div
        class="MuiBox-root MuiBox-root-310 makeStyles-filterStatusContainer-233"
      >
        <div
          class="MuiBox-root MuiBox-root-311 makeStyles-filterStatusContainer-233 makeStyles-autocompleteFilterStatus-250"
        >
          <div
            class="makeStyles-floatingFilterWrapper-239 makeStyles-filterWrapper-238 makeStyles-buttonContainer-313 floatingFilterWrapper"
          >
            <span
              class="makeStyles-floatingFilterButton-240"
              data-testid="controlled-button"
              tabindex="-1"
            >
              <button
                aria-controls="false"
                aria-haspopup="true"
                class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttonFilter-248"
                data-testid="button-filter-open"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  All Projects
                  <span
                    class="MuiButton-endIcon MuiButton-iconSizeMedium"
                  >
                    <svg
                      class="makeStyles-dropdownIcon-242"
                    >
                      expand.svg
                    </svg>
                  </span>
                </span>
              </button>
            </span>
          </div>
        </div>
      </div>
    </div>
    <table
      aria-label="device-list"
      class="MuiTable-root"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Name
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Certification ID
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Completion Date
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Expiration Date
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Project
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-317"
          data-testid="certification-list-row"
          style="cursor: pointer;"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            test certification-g
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            1892132
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
            data-testid="row-completion"
          >
            January 01, 2001
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
            data-testid="row-expiration"
          >
            January 01, 2001
             
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-noTopCell-257 makeStyles-threeDotsWrapper-246"
            data-testid="row-project"
          >
            <a
              class="Link-root-318"
              href="/projects/detail/9164e4c4-6521-47bb-97fd-c75ac02b2cf2"
              title="View Project detail"
            >
              Project Name
            </a>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
            data-testid="update-delete-cell"
          />
        </tr>
      </tbody>
    </table>
    <nav
      aria-label="pagination navigation"
      class="MuiPagination-root  makeStyles-pagination-319"
      data-testid="pagination"
    >
      <ul
        class="MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiPaginationItem-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-current="true"
            aria-label="page 1"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-selected"
            tabindex="0"
            type="button"
          >
            1
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiPaginationItem-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </button>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`CertificationTab should render detail loading 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="makeStyles-tableWrapper-519"
  >
    <div
      class="makeStyles-filterActionsContainer-530 makeStyles-filterActionsContainerPadding-532"
    >
      <div
        class="MuiBox-root MuiBox-root-610 makeStyles-filterStatusContainer-533"
      >
        <div
          class="MuiBox-root MuiBox-root-611 makeStyles-filterStatusContainer-533 makeStyles-autocompleteFilterStatus-550"
        >
          <div
            class="makeStyles-floatingFilterWrapper-539 makeStyles-filterWrapper-538 makeStyles-buttonContainer-613 floatingFilterWrapper"
          >
            <span
              class="makeStyles-floatingFilterButton-540"
              data-testid="controlled-button"
              tabindex="-1"
            >
              <button
                aria-controls="false"
                aria-haspopup="true"
                class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttonFilter-548"
                data-testid="button-filter-open"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  All Projects
                  <span
                    class="MuiButton-endIcon MuiButton-iconSizeMedium"
                  >
                    <svg
                      class="makeStyles-dropdownIcon-542"
                    >
                      expand.svg
                    </svg>
                  </span>
                </span>
              </button>
            </span>
          </div>
        </div>
      </div>
    </div>
    <table
      aria-label="device-list"
      class="MuiTable-root"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Name
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Certification ID
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Completion Date
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Expiration Date
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Project
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-617"
          data-testid="certification-list-row"
          style="cursor: pointer;"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            test certification-g
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            1892132
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
            data-testid="row-completion"
          >
            January 01, 2001
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
            data-testid="row-expiration"
          >
            January 01, 2001
             
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-noTopCell-557 makeStyles-threeDotsWrapper-546"
            data-testid="row-project"
          >
            <a
              class="Link-root-618"
              href="/projects/detail/9164e4c4-6521-47bb-97fd-c75ac02b2cf2"
              title="View Project detail"
            >
              Project Name
            </a>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
            data-testid="update-delete-cell"
          />
        </tr>
      </tbody>
    </table>
    <nav
      aria-label="pagination navigation"
      class="MuiPagination-root  makeStyles-pagination-619"
      data-testid="pagination"
    >
      <ul
        class="MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiPaginationItem-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-current="true"
            aria-label="page 1"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-selected"
            tabindex="0"
            type="button"
          >
            1
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiPaginationItem-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </button>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`CertificationTab should render loading 1`] = `
<div>
  <div
    class="makeStyles-tableWrapper-421"
  >
    <div
      class="makeStyles-filterActionsContainer-432 makeStyles-filterActionsContainerPadding-434"
    >
      <div
        class="MuiBox-root MuiBox-root-512 makeStyles-filterStatusContainer-435"
      >
        <div
          class="MuiBox-root MuiBox-root-513 makeStyles-filterStatusContainer-435 makeStyles-autocompleteFilterStatus-452"
        >
          <div
            class="makeStyles-floatingFilterWrapper-441 makeStyles-filterWrapper-440 makeStyles-buttonContainer-515 floatingFilterWrapper"
          >
            <span
              class="makeStyles-floatingFilterButton-442"
              data-testid="controlled-button"
              tabindex="-1"
            >
              <button
                aria-controls="false"
                aria-haspopup="true"
                class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttonFilter-450"
                data-testid="button-filter-open"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  All Projects
                  <span
                    class="MuiButton-endIcon MuiButton-iconSizeMedium"
                  >
                    <svg
                      class="makeStyles-dropdownIcon-444"
                    >
                      expand.svg
                    </svg>
                  </span>
                </span>
              </button>
            </span>
          </div>
        </div>
      </div>
    </div>
    Loading...
  </div>
</div>
`;

exports[`CertificationTab should render with secondary values 1`] = `
<div>
  <div
    class="makeStyles-tableWrapper-320"
  >
    <div
      class="makeStyles-filterActionsContainer-331 makeStyles-filterActionsContainerPadding-333"
    >
      <div
        class="MuiBox-root MuiBox-root-411 makeStyles-filterStatusContainer-334"
      >
        <div
          class="MuiBox-root MuiBox-root-412 makeStyles-filterStatusContainer-334 makeStyles-autocompleteFilterStatus-351"
        >
          <div
            class="makeStyles-floatingFilterWrapper-340 makeStyles-filterWrapper-339 makeStyles-buttonContainer-414 floatingFilterWrapper"
          >
            <span
              class="makeStyles-floatingFilterButton-341"
              data-testid="controlled-button"
              tabindex="-1"
            >
              <button
                aria-controls="false"
                aria-haspopup="true"
                class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttonFilter-349"
                data-testid="button-filter-open"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  All Projects
                  <span
                    class="MuiButton-endIcon MuiButton-iconSizeMedium"
                  >
                    <svg
                      class="makeStyles-dropdownIcon-343"
                    >
                      expand.svg
                    </svg>
                  </span>
                </span>
              </button>
            </span>
          </div>
        </div>
      </div>
    </div>
    <table
      aria-label="device-list"
      class="MuiTable-root"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Name
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Certification ID
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Completion Date
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Expiration Date
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Project
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-418"
          data-testid="certification-list-row"
          style="cursor: pointer;"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            test certification-g
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            1892132
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
            data-testid="row-completion"
          >
            January 01, 2001
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
            data-testid="row-expiration"
          >
            January 01, 2001
             
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-noTopCell-358 makeStyles-threeDotsWrapper-347"
            data-testid="row-project"
          >
            <a
              class="Link-root-419"
              href="/projects/detail/9164e4c4-6521-47bb-97fd-c75ac02b2cf2"
              title="View Project detail"
            >
              Project Name
            </a>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
            data-testid="update-delete-cell"
          />
        </tr>
        <tr
          class="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-418"
          data-testid="certification-list-row"
          style="cursor: pointer;"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            test other cert
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            329812
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
            data-testid="row-completion"
          >
            -
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
            data-testid="row-expiration"
          >
            -
             
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-noTopCell-358 makeStyles-threeDotsWrapper-347"
            data-testid="row-project"
          >
            <a
              class="Link-root-419"
              href="/projects/detail/8aa1c4c4-6521-47bb-27fd-af5ac02b2cd8"
              title="View Project detail"
            >
              Project Name 2
            </a>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
            data-testid="update-delete-cell"
          />
        </tr>
        <tr
          class="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-418"
          data-testid="certification-list-row"
          style="cursor: pointer;"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            test other cert
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            -
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
            data-testid="row-completion"
          >
            -
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
            data-testid="row-expiration"
          >
            -
             
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-noTopCell-358 makeStyles-threeDotsWrapper-347"
            data-testid="row-project"
          >
            <a
              class="Link-root-419"
              href="/projects/detail/8aa1c4c4-6521-47bb-27fd-af5ac02b2cd8"
              title="View Project detail"
            >
              Project Name 2
            </a>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
            data-testid="update-delete-cell"
          />
        </tr>
      </tbody>
    </table>
    <nav
      aria-label="pagination navigation"
      class="MuiPagination-root  makeStyles-pagination-420"
      data-testid="pagination"
    >
      <ul
        class="MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiPaginationItem-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-current="true"
            aria-label="page 1"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-selected"
            tabindex="0"
            type="button"
          >
            1
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiPaginationItem-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </button>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
